### PR TITLE
lite2: validate TrustOptions, add NewClientFromTrustedStore

### PR DESCRIPTION
**Pull request by melekes**
and make maxRetryAttempts an option.

* [ ]~~ Referenced an issue explaining the need for the change~~
* [ ] ~~Updated all relevant documentation in docs~~
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [ ] ~~Updated CHANGELOG_PENDING.md~~

### CODE REVIEW

File: lite2/client.go [hunk 1]
```diff
@@ -43,6 +43,20 @@ type TrustOptions struct {
...
 
+// ValidateBasic performs basic validation.
+func (opts TrustOptions) ValidateBasic() error {
+	if opts.Period <= 0 {
+		return errors.New("negative or zero period")
+	}
+	if opts.Height <= 0 {
+		return errors.New("negative or zero height")
+	}
+	if len(opts.Hash) == 0 {
```
Comment (melekes) [inline lite2/client.go]:
```suggestion
	if len(opts.Hash) != tmhash.Size {
```

File: lite2/client.go [hunk 2]
```diff
@@ -128,6 +150,7 @@ type Client struct {
 	trustingPeriod   time.Duration // see TrustOptions.Period
 	verificationMode mode
 	trustLevel       tmmath.Fraction
+	maxRetryAttempts int // see MaxRetryAttempts option
```
Comment (marbar3778) [inline lite2/client.go]: should we start being more strict on int types? this may be better suited as an int32 or uint32 possibly, maybe even smaller.

Comment (melekes) [reply lite2/client.go]: good idea

File: lite2/client_test.go [hunk 3]
```diff
@@ -987,3 +987,49 @@ func TestProvider_TrustedHeaderFetchesMissingHeader(t *testing.T) {
...
+		chainID,
+		1*time.Hour,
+		primary,
+		[]provider.Provider{primary},
+		db,
+	)
+	require.NoError(t, err)
+
+	// 3) Check header exists through the lite clients eyes
+	h, err := c.TrustedHeader(1, bTime.Add(1*time.Second))
```
Comment (melekes) [inline lite2/client_test.go]: how do you know it wasn't fetched from primary?

Comment (cmwaters) [reply lite2/client_test.go]: does the lite client automatically update from the primary on start up?

Comment (melekes) [reply lite2/client_test.go]: no, but TrustedHeader can fetch missing headers, so how do you now they are not coming from primary?

Comment (cmwaters) [reply lite2/client_test.go]: but trusted header would return an error if the database was empty: "no headers exist"

Comment (melekes) [reply lite2/client_test.go]: https://github.com/tendermint/tendermint/pull/4374/commits/3bc2b1c5e20078e6b7a00c14dee6e720fbbc3ab9

Comment (melekes) [reply lite2/client_test.go]: You're right. Anyway, it never a bad idea to double check 😄

**Pull request merged**